### ISSUE TITLE
[Bug]: I can add an extra char to `ciphertext`, `salt` and `initial vector` on UTC/JSON File and still derive the correct Address

### ISSUE THREAD

**Issue by seaona** [MetaMask/metamask-extension#15182]
### Describe the bug

**Problem**: I can add an extra char to `ciphertext`, `salt` and `initial vector` on the UTC/JSON File and still derive the correct Address.

![image](https://user-images.githubusercontent.com/54408225/178302988-57355cc4-3b48-46ec-9345-58c2e38b162b.png)

**Extra Info**: https://cryptobook.nakov.com/symmetric-key-ciphers/ethereum-wallet-encryption

### Steps to reproduce

1. Unlock MM
2. Click Import Account
3. Select JSON File type
4. On a new window generate an address (i.e. [here](https://vanity-eth.tk/)) and save it as JSON/UTC
5. Load the file and include the password - you will see that Account is loaded correctly.
6. Now modify the file and add one more char to the following params:
- `salt`
- `ciphertext`
- `iv`
7.Import the Account again and see that you can successfully do it and the same account as before is loaded.

**Note**: I'm not sure how this is possible as modifying the ciphertext would mean modifying the private key. Also modifying `salt` should imply some changes too on the derivation key. Still not sure if this would have any other security implications

### Version

10.16.2

### Browser

Chrome

### Operating system

Linux

**Comment by AlexJupiter** [comment]
@danroc to investigate this issue to see how severe this could be.

**Comment by danroc** [comment]
I can only partially reproduce it:

1. If I add one single character, indeed I can import the key.
2. If I add two or more characters, I get an error saying that the password may be incorrect or that the IV's length is invalid.

This is the expected behavior. Those are hexadecimal strings, which requires an even number of character to be valid. And an odd number of characters will be truncated as stated on [Node's spec][1]:

> Data truncation may occur when decoding strings that do not exclusively consist of an even number of hexadecimal characters. See below for an example.

So, to summarize:
- Point 1 is expected since the conversion from hexadecimal will truncate the last character.
- Point 2 fails will the correct message.

@seaona, do you still have a wallet capable of reproducing the error when 2 or more characters are added to the fields?

[1]: https://github.com/nodejs/node/blob/main/doc/api/buffer.md#buffers-and-character-encodings
